java/transforms-by-suitability/src/main/java/com/esri/arcgisruntime/sample/transformsbysuitability/MainActivity.java

```diff
@@ -0,0 +1,326 @@
+      transformationsBySuitability = TransformationCatalog.getTransformationsBySuitability(inputSr, outputSr);
+    }
+    //[DocRef: END]
+
+    //[DocRef: Name=Get default transform, Category=Fundamentals, Topic=Spatial references]
+    DatumTransformation defaultTransform = TransformationCatalog.getTransformation(inputSr, outputSr);
+    //[DocRef: END]
+
+    // Update user interface with list of transformations to show to user
+    updateTransformsList(transformationsBySuitability, defaultTransform);
+  }
+
+  /**
+   * Update the ArrayList of transformations used by the adapter and notify the adapter of changes.
+   *
+   * @param transforms a List of DatumTransformations to update the adapter with.
+   */
+  private void updateTransformsList(List<DatumTransformation> transforms, DatumTransformation defaultValue) {
+    // Clear any existing values from the ArrayList used by the adapter,
+    mTransformValues.clear();
+
+    // Fill the ArrayList with new values
+    mTransformValues.addAll(transforms);
+
+    // Set the default value on the adapter and notify the adapter that content has changed
+    mTransformAdapter.setDefaultTransformation(defaultValue);
+    mTransformAdapter.notifyDataSetChanged();
+  }
+
+  /**
+   * Add a new Graphic to a GraphicsOverlay in the MapView contained in the layout. Creates and adds a
+   * new GraphicsOverlay to the MapView's collection, if required.
+   *
+   * @param graphicGeometry a Point to be used as the Geometry of the new Graphic
+   * @param color           integer representing the color of the new Symbol
+   * @return the Graphic which was added to an overlay
+   */
+  private Graphic addGraphic(Point graphicGeometry, int color, SimpleMarkerSymbol.Style style) {
+    if (mMapView.getGraphicsOverlays().size() < 1) {
+      mMapView.getGraphicsOverlays().add(new GraphicsOverlay());
+    }
+    SimpleMarkerSymbol sym = new SimpleMarkerSymbol(style, color, 15.0f);
+    Graphic g = new Graphic(graphicGeometry, sym);
+    mMapView.getGraphicsOverlays().get(0).getGraphics().add(g);
+    return g;
+  }
+
+  /**
+   * If the MapView has a GraphicsOverlay, and the Graphic showing the geometry projected with a specific
+   * transformation is set, then remove it from the GraphicsOverlay, and set it to null;
+   */
+  private void removeProjectedGeometryGraphic() {
+    // Remove graphic showing the projected geometry, as the selected transformation is not usable.
+    if ((mMapView.getGraphicsOverlays().size() > 0) && (projectedGraphic != null)) {
+      if (mMapView.getGraphicsOverlays().get(0).getGraphics().size() == 2) {
+        mMapView.getGraphicsOverlays().get(0).getGraphics().remove(projectedGraphic);
+        projectedGraphic = null;
+      }
+    }
+    return;
+  }
+
+  /**
+   * Check if permissions for local storage access have been granted. If so, set the projection engine data
+   * directory and populate the list of transformations. If not, request permissions from user.
+   */
```

Comment:
End on a full stop instead of a semi colon

Reply:
I don't understand your comment - the method comment here does not have a semi colon.

Reply:
Hmm, thats odd. When I was reading through the code. The method header ended in semi-colon rather than a full stop. I must have hit something myself as I was reading through it. Nevermind!